include/circt/Dialect/Seq/SeqOps.td

```diff
@@ -93,6 +93,39 @@ def CompRegClockEnabledOp : SeqOp<"compreg.ce",
   ];
 }
 
+def ShiftRegOp : SeqOp<"shiftreg", [
+  Pure,
+  Clocked,
+  Resettable,
+  AllTypesMatch<["input", "data"]>,
+  SameVariadicOperandSize,
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+  DeclareOpInterfaceMethods<InnerSymbol, ["getTargetResultIndex"]>
+]> {
+  let summary = "Shift register";
+  let description = [{
+    The `seq.shiftreg` op represents a shift register. It takes the input
+    value and shifts it every cycle when `clockEnable` is asserted.
+    The `reset` and `resetValue` operands are optional and if present, every
+    entry in the shift register will be initialized to `resetValue` on the
+    rising edge of the `reset` signal.
+  }];
+
+  let arguments = (ins
+    ConfinedAttr<I64Attr, [IntMinValue<1>]>:$size,
+    AnyType:$input,
+    I1:$clk,
+    I1:$clockEnable,
+    StrAttr:$name,
```

Comment:
Should name be optional?

Reply:
If we're following the style of the remainder of the seq dialect ops, then no - `name` is set during parsing if not provided via.  `setNameFromResult`.

Reply:
Changed to `OptionalAttr` now that the assembly format follows that of the other seq ops.